pkg/sandbox/infra.go

```diff
@@ -68,3 +85,100 @@ func PauseCommand(cfg *libconfig.Config, image *v1.Image) ([]string, error) {
 	cmd = append(cmd, image.Config.Cmd...)
 	return cmd, nil
 }
+
+func (s *sandbox) createResolvPath(podContainer *storage.ContainerInfo) (retErr error) {
+	// set DNS options
+	if s.config.DNSConfig == nil {
+		return nil
+	}
+
+	dnsServers := s.config.DNSConfig.Servers
+	dnsSearches := s.config.DNSConfig.Searches
+	dnsOptions := s.config.DNSConfig.Options
+	s.resolvPath = fmt.Sprintf("%s/resolv.conf", podContainer.RunDir)
+	err := ParseDNSOptions(dnsServers, dnsSearches, dnsOptions, s.resolvPath)
+	defer func() {
+		if retErr != nil {
+			if err := os.RemoveAll(s.resolvPath); err != nil {
+				retErr = errors.Wrapf(retErr, "failed to remove resolvPath after failing to create it")
+			}
+		}
+	}()
+	if err != nil {
+		return err
+	}
+
+	if err := label.Relabel(s.resolvPath, podContainer.MountLabel, false); err != nil && !errors.Is(err, unix.ENOTSUP) {
+		return err
+	}
+	mnt := spec.Mount{
+		Type:        "bind",
+		Source:      s.resolvPath,
+		Destination: "/etc/resolv.conf",
+		Options:     []string{"ro", "bind", "nodev", "nosuid", "noexec"},
+	}
+	s.infra.Spec().AddMount(mnt)
+	return nil
+}
+
+func ParseDNSOptions(servers, searches, options []string, path string) (retErr error) {
+	nServers := len(servers)
+	nSearches := len(searches)
+	nOptions := len(options)
+	if nServers == 0 && nSearches == 0 && nOptions == 0 {
+		return copyFile("/etc/resolv.conf", path)
+	}
+
+	if nSearches > maxDNSSearches {
+		return fmt.Errorf("DNSOption.Searches has more than 6 domains")
```

Comment:
In case it ever changes out from underneath you...
```suggestion
		return fmt.Errorf("DNSOption.Searches has more than %d domains", maxDNSSearches)
```

Reply:
nice catch!